api/cpp/esp-idf/slint/README.md

```diff
@@ -11,81 +11,110 @@ It has been tested on ESP32-S3 devices.
-
-## Usage
-
-By using this component, the `Slint::Slint` CMake target is linked to your application and you can access the entire functionality of the
-[Slint C++ API](https://slint.dev/docs/cpp).
-
-In addition, this component provides the `slint-esp.h` header file, which provides a a function to initialize the Slint backend
-that must be called before setting up the UI.
-Behind the scene, it will use `esp_lcd_panel_draw_bitmap` to render to the screen,
-and is base on [ESP LCD Touch](https://components.espressif.com/components/espressif/esp_lcd_touch) for the touch events.
-
-Use this platform implementation by instantiating it with a `esp_lcd_panel_handle_t`, an optional `esp_lcd_touch_handle_t`, and a pointer to one
-or two `slint::platform::Rgb565Pixel` frame buffers. Next, register the instance with the Slint run-time library by calling `slint_esp_init()`:
-
-```cpp
-#include "slint-esp.h"
-// ...
-
-
-// Initialize the panel and touch handle
-esp_lcd_panel_handle_t panel_handle = NULL;
-bsp_display_new(&bsp_disp_cfg, &panel_handle, &io_handle);
-bsp_display_backlight_on();
-esp_lcd_touch_handle_t touch_handle = NULL;
-bsp_touch_new(&bsp_touch_cfg, &touch_handle);
-
-// Allocate a frame buffer
-static std::vector<slint::platform::Rgb565Pixel> buffer(BSP_LCD_H_RES * BSP_LCD_V_RES);
-
-// initialize the Slint ESP backend
-slint_esp_init(slint::PhysicalSize({ BSP_LCD_H_RES, BSP_LCD_V_RES }), panel_handle,
-                                   touch_handle, buffer);
-
+import { VerticalBox, AboutSlint } from "std-widgets.slint";
+export component AppWindow inherits Window {
+    VerticalBox {
+        AboutSlint {}
+        Text {
+            text: "Hello World";
+            font-size: 18px;
+            horizontal-alignment: center;
+        }
+    }
+}
 ```
-
-Alternatively, you can implement your own sub-class of `slint::platform::Platform` to drive the screen and handle input events.
-
-Next, integrate your `.slint` files into the build by compiling them to C++ and linking them to your component:
-
+1. Edit `main/CMakeLists.txt` to adjust for the new `slint-hello-world.cpp`, add `slint` as required component,
+   and instruction the build system to compile `appwindow.slint` to `appwindow.h`. The file should look like this:
 ```cmake
-slint_target_sources(${COMPONENT_LIB} my_application_ui.slint)
-```
-
-Instantiate the Slint component by including the generated header file (`my_application_ui.h` in the above example), call `create()`
-on the generated class and `run()` to spin the event loop:
-
-```cpp
-#include "my_application_ui.h"
-
-/// ...
-
-auto my_application_ui = MainWindow::create();
-
-my_application_ui->run();
+idf_component_register(SRCS "slint-hello-world.cpp" INCLUDE_DIRS "." REQUIRES slint)
+slint_target_sources(${COMPONENT_LIB} appwindow.slint)
 ```
+1. Open the configuration editor with `idf.py menuconfig`:
+    1. Change the stack size under `Component config --> ESP System Settings --> Main task stack size` to `13584`.
```

Comment:
this is some random number i chose (just added "1" to the default).
I wonder if we need so much.

Reply:
Ahh, good to know you "algorithm" :). I'll try with the default for this hello world and add a section to the FAQ about bumping it.